metrics: m.Compact.Duration decreases and causes panic

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/15826599?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&expandBuildTestsSection=true

```
panic: Counters should not decrease [recovered]
  panic: Counters should not decrease [recovered]
  panic: Counters should not decrease [recovered]
  panic: Counters should not decrease
goroutine 28427 [running]:
testing.tRunner.func1.2({0x83c1b20, 0xc8f0e30})
  GOROOT/src/testing/testing.go:1631 +0x3f7
testing.tRunner.func1()
  GOROOT/src/testing/testing.go:1634 +0x6b6
panic({0x83c1b20?, 0xc8f0e30?})
  GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
  github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:133 +0x4fa
panic({0x83c1b20?, 0xc8f0e30?})
  GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).Start.func1()
  github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:393 +0x99
panic({0x83c1b20?, 0xc8f0e30?})
  GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/metric.(*Counter).Update(0xc007584f00, 0x5735f786a)
  github.com/cockroachdb/cockroach/pkg/util/metric/pkg/util/metric/metric.go:755 +0x6d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*StoreMetrics).updateEngineMetrics(_, {0xc013740d08, {0x273f4dd, 0x93eae, 0x480b4c1b, 0xdf8, 0x1e79, 0xdf8, 0x289c}, {0x9ab, ...}, ...})
  github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/metrics.go:3765 +0x11bb
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).computeMetrics(_, {_, _})
  github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:3478 +0x1d8
```
https://github.com/cockroachdb/cockroach/blob/82cf31008da2a52a5d9de42b10d90393f753266a/pkg/kv/kvserver/metrics.go#L3765

--- a/db.go
+++ b/db.go
@@ -1944,7 +1944,7 @@ func (d *DB) Metrics() *Metrics {
 	metrics.Compact.MarkedFiles = vers.Stats.MarkedForCompaction
 	metrics.Compact.Duration = d.mu.compact.duration
 	for c := range d.mu.compact.inProgress {
-		if c.kind != compactionKindFlush {
+		if c.kind != compactionKindFlush && c.kind != compactionKindIngestedFlushable {
 			metrics.Compact.Duration += d.timeNow().Sub(c.beganAt)
 		}
 	}

